jira_agile_metrics/calculators/defects.py

```diff
@@ -77,120 +90,198 @@ def run(self):
+
+
+
         if window:
             breakdown = breakdown[-window:]
-        
+
         if len(breakdown.index) == 0 or len(breakdown.columns) == 0:
             logger.warning("Cannot draw defects by priority chart with zero items")
             return
 
         fig, ax = plt.subplots()
-        
+
         breakdown.plot.bar(ax=ax, stacked=True)
-        
+
         if self.settings['defects_by_priority_chart_title']:
             ax.set_title(self.settings['defects_by_priority_chart_title'])
 
         ax.legend(loc='center left', bbox_to_anchor=(1, 0.5))
-        ax.set_xlabel("Month", labelpad=20)
-        ax.set_ylabel("Number of items", labelpad=10)
+        ax.set_xlabel(x_label, labelpad=20)
+        ax.set_ylabel("Number of Defects", labelpad=10)
+
+        if x_label =='Month':
+            labels = [d.strftime("%b %y") for d in breakdown.index]
+            ax.set_xticklabels(labels, rotation=90, size='small')
+            ax.set_xlabel("Month", labelpad=20)
 
-        labels = [d.strftime("%b %y") for d in breakdown.index]
-        ax.set_xticklabels(labels, rotation=90, size='small')
 
-        set_chart_style()
+        # set_chart_style()
 
         # Write file
         logger.info("Writing defects by priority chart to %s", output_file)
         fig.savefig(output_file, bbox_inches='tight', dpi=300)
         plt.close(fig)
-    
-    def write_defects_by_type_chart(self, chart_data, output_file):
+
+    def write_defects_by_type_chart(self, chart_data, output_file,x_label):
         window = self.settings['defects_window']
         type_values = self.settings['defects_type_values']
+       
+        if x_label !='Month':
+            breakdown = chart_data.pivot_table(index=['fixversion'],values='key',columns=['type'],aggfunc='count')
+        else:
+            breakdown = breakdown_by_month(chart_data, 'created', 'resolved', 'key', 'type', type_values)
 
-        breakdown = breakdown_by_month(chart_data, 'created', 'resolved', 'key', 'type', type_values)
-        
         if window:
             breakdown = breakdown[-window:]
-        
+
         if len(breakdown.index) == 0 or len(breakdown.columns) == 0:
             logger.warning("Cannot draw defects by type chart with zero items")
             return
 
         fig, ax = plt.subplots()
-        
+
         breakdown.plot.bar(ax=ax, stacked=True)
-        
+
         if self.settings['defects_by_type_chart_title']:
             ax.set_title(self.settings['defects_by_type_chart_title'])
 
         ax.legend(loc='center left', bbox_to_anchor=(1, 0.5))
-        ax.set_xlabel("Month", labelpad=20)
-        ax.set_ylabel("Number of items", labelpad=10)
+        ax.set_xlabel(x_label, labelpad=20)
+        ax.set_ylabel("Number of Defects", labelpad=10)
 
-        labels = [d.strftime("%b %y") for d in breakdown.index]
-        ax.set_xticklabels(labels, rotation=90, size='small')
+        if  x_label =='Month':
```

Comment:
Comparator again

Reply:
I will initialise it with a boolean in the start so that it can be used across.